engine/cache/manager.go

```diff
@@ -510,7 +516,7 @@ func (m *manager) descriptorProviderPair(layerMetadata remotecache.CacheLayer) (
 
 type Manager interface {
 	solver.CacheManager
-	StartCacheMountSynchronization(context.Context) error
+	SynchronizeCacheMounts(context.Context, []string) error
```

Comment:
`DownloadCacheMounts` sounds better for sure. As for `UploadCacheMounts`, having it on the interface would make it more clear what the manager is supposed to do. I would like to keep the call to upload on `Close` as well, to be sure that cache mounts are uploaded before exiting. WDYT? (maybe with a `once` so that we don't do it again in case it was already done?)

Reply:
Yeah, that sounds good to me - but `Close` is also *only* called in `cmd/engine/main.go`, so shouldn't be that hard to add it right before then.